### PR TITLE
FI-888 fix end range problem and add unit tests

### PR DESCRIPTION
**Pull request by czh-orz**
This PR fixes an issue when getting a value to search by when the period has no start date. It also adds more unit tests for getting a search value from a period .

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-888
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: test/unit/sequence_base_test.rb [hunk 1]
```diff
@@ -123,14 +123,50 @@ def setup
...
+          expected: 'gt2019-12-31T00:00:00+00:00'
+        },
+        {
+          element: FHIR::Period.new('end' => '2020'),
+          expected: 'lt2021-01-01T23:59:59+00:00'
+        },
+        {
+          element: FHIR::Period.new('end' => '2020'),
+          expected: 'lt2021-01-01T23:59:59+00:00'
+        },
```
Comment (radamson) [inline test/unit/sequence_base_test.rb:142]: Looks like this case is duplicated
```suggestion
```

**Comment by radamson** [comment]
Overall looks good! Tested it out and the `get_fhir_datetime_range`addition causes the dates to be incremented with the correct granularity.

File: test/unit/sequence_base_test.rb (hunk 1 above)
Comment (czh-orz) [reply test/unit/sequence_base_test.rb]: Yep, fixed.
